frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import { PromiseComponent, PromiseComponentState } from '@console/internal/components/utils';
+
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory';
+
+import { k8sPatch } from '@console/internal/module/k8s';
+
+import { getPxeBootPatch } from 'kubevirt-web-ui-components';
+import { V1VirtualMachine } from 'kubevirt-typescript-api/esm';
+import { getName, getNamespace } from '@console/shared';
+import { VirtualMachineModel } from '../../models';
+
+class StartStopVmModal extends PromiseComponent<StartStopVmModalProps, PromiseComponentState> {
+  submit = (event) => {
+    event.preventDefault();
+    const { vm } = this.props;
+
+    const patches = [];
+
+    // handle PXE boot
+    if (this.props.start) {
+      const pxePatch = getPxeBootPatch(vm);
+      patches.push(...pxePatch);
+    }
+
+    patches.push({
+      op: 'replace',
+      path: '/spec/running',
+      value: this.props.start,
+    });
+
+    const promise = k8sPatch(VirtualMachineModel, vm, patches);
+    return this.handlePromise(promise).then(this.props.close);
+  };
+
+  render() {
+    const { vm } = this.props;
+    const action = this.props.start ? 'Start' : 'Stop';
+    return (
+      <form onSubmit={this.submit} name="form" className="modal-content">
+        <ModalTitle>{action} Virtual Machine</ModalTitle>
+        <ModalBody>
+          Are you sure you want to {action.toLowerCase()} <strong>{getName(vm)}</strong>
+          <span>
+            {' '}
+            in namespace <strong>{getNamespace(vm)}</strong>?
+          </span>
+        </ModalBody>
+        <ModalSubmitFooter
+          errorMessage={this.state.errorMessage}
+          inProgress={this.state.inProgress}
+          submitText={action}
+          cancel={this.props.cancel}
+        />
+      </form>
+    );
+  }
+}
+
+type StartStopVmModalProps = {
+  vm: V1VirtualMachine;
+  start: boolean;
+  cancel: (e?: Event) => void;
+  close: (e?: Event) => void;
```

Comment:
I don't see us passing the event to these callbacks in the implementation. We should take it out if we're not using it.